src/requirements.ts

```diff
@@ -36,28 +37,33 @@ export async function resolveRequirements(): Promise<RequirementsData> {
   return Promise.resolve({ java_home: java_home, java_version: javaVersion });
 }
 
-function checkJavaRuntime(): Promise<any> {
+function checkJavaRuntime(): Promise<string> {
   return new Promise((resolve, reject) => {
     let source: string;
     let javaHome: string = readJavaConfig();
     if (javaHome) {
-      source = "The 'sonarlint.ls.javaHome' variable defined in VS Code settings";
+      source = "'sonarlint.ls.javaHome' variable defined in VS Code settings";
     } else {
       javaHome = process.env['JDK_HOME'];
       if (javaHome) {
-        source = 'The JDK_HOME environment variable';
+        source = 'JDK_HOME environment variable';
       } else {
         javaHome = process.env['JAVA_HOME'];
-        source = 'The JAVA_HOME environment variable';
+        source = 'JAVA_HOME environment variable';
       }
     }
     if (javaHome) {
       javaHome = expandHomeDir(javaHome);
       if (!pathExists.sync(javaHome)) {
-        openJREDownload(reject, source + ' points to a missing folder');
-      }
-      if (!pathExists.sync(path.resolve(javaHome, 'bin', JAVA_FILENAME))) {
-        openJREDownload(reject, source + ' does not point to a JRE.');
+        invalidJavaHome(reject, `The ${source} points to a missing or inaccessible folder (${javaHome})`);
+      } else if (!pathExists.sync(path.resolve(javaHome, 'bin', JAVA_FILENAME))) {
```

Comment:
With this check we cannot use Windows symbolic link path, for instance: `C:\ProgramData\Oracle\Java\javapath` or `C:\Program Files (x86)\Common Files\Oracle\Java\javapath` which force us to update vscode configuration each team we update the JRE